src/promise/Promise.php

```diff
@@ -23,10 +23,11 @@
 
 namespace pocketmine\promise;
 
+use function count;
 use function spl_object_id;
 
 /**
- * @phpstan-template TValue
+ * @phpstan-template-covariant TValue
```

Comment:
Why is this changed?

Reply:
Without, PHPstan return: 
```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   src/Server.php                                                                                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  :595   Parameter #1 $promises of static method pocketmine\promise\Promise<mixed>::all() expects array<pocketmine\promise\Promise<mixed>>, array<int, pocketmine\promise\Promise<null>> given.  
         💡 Template type TValue on class pocketmine\promise\Promise is not covariant. Learn more: https://phpstan.org/blog/whats-up-with-template-covariant                                     
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ```

Reply:
I see. I suppose it makes sense in this case considering that `Promise` is a container class.

Reply:
I personally don't love template-covariant as I find it very confusing. But in this case I think it's probably OK.